### DISCUSSION TITLE
pip install the CPU version?

### DISCUSSION THREAD

**Discussion by johann-petrak** [vllm-project/vllm#8090] [Q&A]
I would like to install the cpu version, but on Ubuntu 22.04 the cmake version is not what the compilation approach requires.

On the other hand the docs tell me that pip-based installation always installs the GPU version which is not usable on my machine. Is there a way to install a pre-compiled CPU wheel for Ubuntu?

Is there any other way to make the CPU version work?

**Comment by dtrifiro** [comment]
I found the following strategy to work quite well:

```bash
VLLM_TARGET_DEVICE=cpu 
    PIP_EXTRA_INDEX_URL=https://download.pytorch.org/whl/cpu \
    pip install https://github.com/vllm-project/vllm
```
